### ISSUE TITLE
--rc-web-gui cannot create/edit configs

### ISSUE THREAD

**Issue by jackitsdotcom** [rclone/rclone#4130]
#### What is the problem you are having with rclone?
When I try and use the `--rc-web-gui` with the `--rc-no-auth` flag, I still get the following errors:

-> `2020/04/15 07:57:58 ERROR : rc: "config/get": error: authentication must be set up on the rc server to use "config/get" or the --rc-no-auth flag must be in use`
No configs can be created.

Once I used `-vv` I saw that the GUI cannot now be run without auth.
`2020/04/15 08:02:04 INFO  : Cannot run Web GUI without authentication, using default auth`

Even without the `--rc-no-auth` flags, I continue to get the authentication error arrowed above.

#### What is your rclone version (output from `rclone version`)
rclone v1.51.0
- os/arch: windows/amd64
- go version: go1.13.7

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)
Windows 10 x64 v1903 build 18362.370

####  Which cloud storage system are you using? (eg Google Drive)
OneDrive for Business

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)
Trying to use the web gui.

#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)

With the `--rc-no-auth` flag:
```
→ C:\Users\[redacted for privacy]› rclone rcd --rc-web-gui --rc-no-auth -vv
2020/04/15 08:06:38 DEBUG : rclone: Version "v1.51.0" starting with parameters ["[redacted for privacy]\\rclone-v1.51.0-windows-amd64\\rclone.exe" "rcd" "--rc-web-gui" "--rc-no-auth" "-vv"] 2020/04/15 08:06:38 NOTICE: Web GUI exists. Update skipped.
2020/04/15 08:06:38 INFO  : Cannot run Web GUI without authentication, using default auth
2020/04/15 08:06:38 INFO  : No username specified. Using default username: gui
2020/04/15 08:06:38 INFO  : No password specified. Using random password: [redacted for privacy]
2020/04/15 08:06:38 NOTICE: Serving Web GUI
2020/04/15 08:06:38 NOTICE: Serving remote control on http://127.0.0.1:[redacted for privacy]/
2020/04/15 08:06:38 DEBUG : login_token "[redacted for privacy]"
2020/04/15 08:06:38 DEBUG : rc: "core/stats": with parameters map[]
2020/04/15 08:06:38 DEBUG : rc: "core/stats": with parameters map[]
2020/04/15 08:06:38 DEBUG : rc: "core/stats": reply map[bytes:0 checks:0 deletes:0 elapsedTime:0 errors:0 fatalError:false retryError:false speed:0 transfers:0]: <nil>
2020/04/15 08:06:38 DEBUG : rc: "core/stats": reply map[bytes:0 checks:0 deletes:0 elapsedTime:0 errors:0 fatalError:false retryError:false speed:0 transfers:0]: <nil>
2020/04/15 08:06:38 DEBUG : rc: "core/stats": with parameters map[]
2020/04/15 08:06:38 DEBUG : rc: "core/stats": reply map[bytes:0 checks:0 deletes:0 elapsedTime:0 errors:0 fatalError:false retryError:false speed:0 transfers:0]: <nil>
2020/04/15 08:06:38 DEBUG : rc: "core/stats": with parameters map[]
2020/04/15 08:06:38 DEBUG : rc: "core/stats": reply map[bytes:0 checks:0 deletes:0 elapsedTime:0 errors:0 fatalError:false retryError:false speed:0 transfers:0]: <nil>
2020/04/15 08:06:38 DEBUG : rc: "core/stats": with parameters map[]
2020/04/15 08:06:38 DEBUG : rc: "core/stats": reply map[bytes:0 checks:0 deletes:0 elapsedTime:0 errors:0 fatalError:false retryError:false speed:0 transfers:0]: <nil>
2020/04/15 08:06:38 DEBUG : rc: "core/stats": with parameters map[]
2020/04/15 08:06:38 DEBUG : rc: "core/stats": reply map[bytes:0 checks:0 deletes:0 elapsedTime:0 errors:0 fatalError:false retryError:false speed:0 transfers:0]: <nil>
2020/04/15 08:06:38 DEBUG : rc: "core/stats": with parameters map[]
2020/04/15 08:06:38 DEBUG : rc: "core/bwlimit": with parameters map[]
2020/04/15 08:06:38 DEBUG : rc: "core/stats": reply map[bytes:0 checks:0 deletes:0 elapsedTime:0 errors:0 fatalError:false retryError:false speed:0 transfers:0]: <nil>
2020/04/15 08:06:38 DEBUG : rc: "core/bwlimit": reply map[bytesPerSecond:-1 rate:off]: <nil>
2020/04/15 08:06:40 ERROR : rc: "config/dump": error: authentication must be set up on the rc server to use "config/dump" or the --rc-no-auth flag must be in use
2020/04/15 08:06:41 ERROR : rc: "config/providers": error: authentication must be set up on the rc server to use "config/providers" or the --rc-no-auth flag must be in use
```

Without the no-auth flag:
```
→ C:\Users\[redacted for privacy]› rclone rcd --rc-web-gui -vv
2020/04/15 08:07:18 DEBUG : rclone: Version "v1.51.0" starting with parameters ["[redacted for privacy]\\rclone-v1.51.0-windows-amd64\\rclone.exe" "rcd" "--rc-web-gui" "-vv"]
2020/04/15 08:07:18 NOTICE: Web GUI exists. Update skipped.
2020/04/15 08:07:18 INFO  : No username specified. Using default username: gui
2020/04/15 08:07:18 INFO  : No password specified. Using random password: [redacted for privacy]
2020/04/15 08:07:18 NOTICE: Serving Web GUI
2020/04/15 08:07:18 NOTICE: Serving remote control on http://127.0.0.1:[redacted for privacy]/
2020/04/15 08:07:18 DEBUG : login_token "[redacted for privacy]"
2020/04/15 08:07:18 DEBUG : rc: "core/stats": with parameters map[]
2020/04/15 08:07:18 DEBUG : rc: "core/stats": reply map[bytes:0 checks:0 deletes:0 elapsedTime:0 errors:0 fatalError:false retryError:false speed:0 transfers:0]: <nil>
2020/04/15 08:07:18 DEBUG : rc: "core/stats": with parameters map[]
2020/04/15 08:07:18 DEBUG : rc: "core/stats": reply map[bytes:0 checks:0 deletes:0 elapsedTime:0 errors:0 fatalError:false retryError:false speed:0 transfers:0]: <nil>
2020/04/15 08:07:18 DEBUG : rc: "core/stats": with parameters map[]
2020/04/15 08:07:18 DEBUG : rc: "core/stats": reply map[bytes:0 checks:0 deletes:0 elapsedTime:0 errors:0 fatalError:false retryError:false speed:0 transfers:0]: <nil>
2020/04/15 08:07:18 DEBUG : rc: "core/bwlimit": with parameters map[]
2020/04/15 08:07:18 DEBUG : rc: "core/bwlimit": reply map[bytesPerSecond:-1 rate:off]: <nil>
2020/04/15 08:07:18 DEBUG : rc: "core/stats": with parameters map[]
2020/04/15 08:07:18 DEBUG : rc: "core/stats": with parameters map[]
2020/04/15 08:07:18 DEBUG : rc: "core/stats": reply map[bytes:0 checks:0 deletes:0 elapsedTime:0 errors:0 fatalError:false retryError:false speed:0 transfers:0]: <nil>
2020/04/15 08:07:18 DEBUG : rc: "core/stats": reply map[bytes:0 checks:0 deletes:0 elapsedTime:0 errors:0 fatalError:false retryError:false speed:0 transfers:0]: <nil>
2020/04/15 08:07:18 DEBUG : rc: "core/stats": with parameters map[]
2020/04/15 08:07:18 DEBUG : rc: "core/stats": reply map[bytes:0 checks:0 deletes:0 elapsedTime:0 errors:0 fatalError:false retryError:false speed:0 transfers:0]: <nil>
2020/04/15 08:07:19 DEBUG : rc: "core/stats": with parameters map[]
2020/04/15 08:07:19 DEBUG : rc: "core/stats": reply map[bytes:0 checks:0 deletes:0 elapsedTime:0 errors:0 fatalError:false retryError:false speed:0 transfers:0]: <nil>
2020/04/15 08:07:20 ERROR : rc: "config/dump": error: authentication must be set up on the rc server to use "config/dump" or the --rc-no-auth flag must be in use
2020/04/15 08:07:21 DEBUG : rc: "core/stats": with parameters map[]
2020/04/15 08:07:21 DEBUG : rc: "core/stats": with parameters map[]
2020/04/15 08:07:21 DEBUG : rc: "core/stats": reply map[bytes:0 checks:0 deletes:0 elapsedTime:0 errors:0 fatalError:false retryError:false speed:0 transfers:0]: <nil>
2020/04/15 08:07:21 DEBUG : rc: "core/stats": reply map[bytes:0 checks:0 deletes:0 elapsedTime:0 errors:0 fatalError:false retryError:false speed:0 transfers:0]: <nil>
2020/04/15 08:07:22 ERROR : rc: "config/providers": error: authentication must be set up on the rc server to use "config/providers" or the --rc-no-auth flag must be in use
```

**Comment by Yexiaoxing** [comment]
What browser are you using?

**Comment by Yexiaoxing** [comment]
Well I think I got the reason. In https://github.com/rclone/rclone/blob/38a4d50e73060e1b486de6dc0caba68fa0a0c0a0/fs/rc/rcserver/rcserver.go#L65 the http server is started before webui set the username and password, causing the internal flag `usingAuth` to be false. But we do use this flag to identify whether the server is auth-enabled.

**Comment by ncw** [comment]
I've merged this @Yexiaoxing 's fix to master - thank you! - now which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.52
